website/docs/cdktf/examples-and-guides/refactoring.mdx

```diff
@@ -0,0 +1,255 @@
+---
+page_title: Refactoring - CDK for Terraform
+description: >-
+  Refactor stacks as your CDKTF application grows in size and complexity.
+---
+
+# Refactoring
+
+You may need to move resources between stacks as your CDK for Terraform (CDKTF) application grows in size and complexity. This guide demonstrates how to refactor your stacks. In general you can change the names of stacks freely without an effect on the synthesized code. If you change the name of a resource this might lead to a re-creation of the resource, especially if you move the resource from one stack to another.
+
+## Moving resources inside a stack
+
+Let's say you have a stack deployed with an aws `S3Bucket` resource and you want to give it another name:
+
+```typescript
+import { Construct } from "constructs";
+import { App, TerraformStack } from "cdktf";
+import { AwsProvider } from "@cdktf/provider-aws/lib/provider";
+import { S3Bucket } from "@cdktf/provider-aws/lib/s3-bucket";
+
+class MyStack extends TerraformStack {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    new AwsProvider(this, "aws", {
+      region: "us-east-1",
+    });
+
+    // Old S3Bucket definition
+    // new S3Bucket(this, "my-bucket", {
+    //   bucketPrefix: "my-bucket",
+    // });
+
+    // New S3Bucket definition
+    new S3Bucket(this, "new-bucket", {
+      bucketPrefix: "my-bucket",
+    });
+  }
+}
+
+const app = new App();
+new MyStack(app, "refactoring-example");
+app.synth();
+```
+
+When you run a `cdktf plan` you might see the following output:
+
+```shell-session
+refactoring-example    # aws_s3_bucket.my-bucket will be destroyed
+                       # (because aws_s3_bucket.my-bucket is not in configuration)
+                       - resource "aws_s3_bucket" "my-bucket" {
+...
+refactoring-example    # aws_s3_bucket.new-bucket (new-bucket) will be created
+                       + resource "aws_s3_bucket" "new-bucket" {
+```
+
+As you can see the old bucket is going to be destroyed and a new one is created to replace it. This can be harmful if there is already state in the resource (e.g. files in the bucket, data in the database, etc).
+
+To avoid the recreation of the resource you can use the [`terraform state mv`](/cli/commands/state/mv) command to move the state before the next apply.
```

Comment:
```suggestion
To avoid the recreation of the resource, use the [`terraform state mv`](/cli/commands/state/mv) command to move the state before the next apply.
```